### PR TITLE
handle a boundary condition error in pmap with err_retry=true

### PR DESCRIPTION
**Pull request by amitmurthy**
Fixes a condition in `pmap` with `err_retry=true`. Where, if the last few tasks threw an error, they were not being retried since the other workers had already exited the processing loop.

### CODE REVIEW

File: test/parallel.jl [hunk 1]
```diff
@@ -223,12 +223,16 @@ if Bool(parse(Int,(get(ENV, "JULIA_TESTFULL", "0"))))
 
     # retry, on error exit
     res = pmap(x->(x=='a') ? error("EXPECTED TEST ERROR. TO BE IGNORED.") : uppercase(x), s; err_retry=true, err_stop=true);
-    @test length(res) < length(ups)
+    if length(res) == length(ups)
+        println("WARNING! Everything finished processing in spite of an error. Though possible it is a typically unexpected condition!")
```
Comment (tkelman) [inline test/parallel.jl]: Why not just `warn` ? "Though possible it is a typically unexpected condition" sounds a little funny, doesn't the fact that this is a warning message already convey that?

Comment (amitmurthy) [inline test/parallel.jl]: I agree. Will change it.

**Comment by vtjnash** [comment]
lgtm (travis failures were unrelated)

File: test/parallel.jl [hunk 2]
```diff
@@ -223,12 +223,19 @@ if Bool(parse(Int,(get(ENV, "JULIA_TESTFULL", "0"))))
 
     # retry, on error exit
     res = pmap(x->(x=='a') ? error("EXPECTED TEST ERROR. TO BE IGNORED.") : uppercase(x), s; err_retry=true, err_stop=true);
-    @test length(res) < length(ups)
+    if length(res) == length(ups)
+        # Under normal circumstances since the first task fails, and err_stop == true, the number of completed tasks must be less
+        # than the number of submitted tasks. Sometimes, if the exception propagated from the failed task is slow, all tasks
+        # may complete before pmap stops.
+        println("WARNING! Very slow propagation of an expected error!")
```
Comment (tkelman) [inline test/parallel.jl]: why `println` and not `warn`?

**Pull request merged**